springwolf-core/src/main/java/io/github/springwolf/core/configuration/properties/PayloadSchemaFormat.java

```diff
@@ -0,0 +1,23 @@
+// SPDX-License-Identifier: Apache-2.0
+package io.github.springwolf.core.configuration.properties;
+
+import io.github.springwolf.asyncapi.v3.model.schema.SchemaFormat;
+
+/**
+ * Enumeration defining the supported payload schema formats, for use in SpringwolfConfigProperties.
+ */
+public enum PayloadSchemaFormat {
```

Comment:
We are not sure if it a good idea to directly map the payload schema into the SchemaFormat enum, which has more types.
It couples the values of this enum to the other other and leaves less room for configuration changes.


Similarly, it feels odd in the switch expression to have an else branch, that can never be reached - at this moment in time, but is required technically for the compiler.

Suggestion:
ASYNCAPI_V3, OPENAPI_V3, OPENAPI_V3_1;

And the PayloadSchemaFormat is used in the code, including the switch expression

Reply:
Yeah, it is something in between.

From our point of view, the mapping from PayloadSchemaFormat to SchemaFormat happened to early. It is probably fine to have the getSchemaFormatForPayloadSchemaFormat method on the enum.

However, most of the code should use (and pass around) the configuration property (PayloadSchemaFormat). Which schema should be used depends on the configuration value (including the enum value) and not on the SchemaFormat enum (which might change, has more configuration options, etc).
Still, the SchemaFormat is used when using the schema format string, but it does not control program flow (switch)

Reply:
I pushed a commit to address/resolve it